### ISSUE TITLE
[0781 PS] (3) PDF Upload Page Skeleton (FE only)

### ISSUE THREAD

**Issue by SamStuckey** [department-of-veterans-affairs/va.gov-team#97066]
## SOFT BLOCKED
We need to have our form JSON boiler plate in place before this work can be _completed_.
- department-of-veterans-affairs/va.gov-team#97063 

### ATTENTION - the design and functionality for upload is different than what exists today
Here's the [component documentation](https://design.va.gov/components/form/file-input)
Here's the [Figma](https://www.figma.com/design/JDFpGLIojfuQwANXScQjqe/VADS-Component-Examples?node-id=1360-85508&p=f&t=acrdmfCegojQYxmq-0)

### Testability

- skeleton
- foundational

### Description

Entry Point: 3.3

Exit Point(s):  [3.1c] Start page - in completed status 

### Acceptance Criteria

- As a user visiting this page
  - I am presented with an option to upload a PDF
  - My PDF upload should upload successfully
  - My PDF upload should be visible upon returning to this page

- As a developer
  - I can confirm that upload PDFs are stored in the correct storage solution

### Expected Deployments
FE:
  - deploy the page skeleton with the PDF upload tool
  - Display [3.3] Upload 0781 page if the user selected the option upload and then continue
  - Once the file is uploaded, the browser presents a dialog to inform the user of the success or failure of the upload. 
  - The uploaded file can then be removed by the user, if necessary. 
  - Additional files can be uploaded, one by one.

**Attempt to upload**
- When a file is being uploaded display the file name the user is attempting to upload 
- When a file is being upload display a progress bar [(design component)](https://design.va.gov/components/progress-bar/)
- When a file is being uploaded it should be checked to meet the guidelines
    - File types allowed:  .pdf, .jpg, .jpeg, .png, .gif, .bmp, or .txt
    - Maximum non-PDF file size: 150MB

 
**Unsuccessful upload**
- If the user attempts to upload a file that is not supported, display an error message, direct focus to the error message

**Successful upload** 
- When the file is successfully uploaded display the file type with the new 0781 form name: 
   "VA Form 21-0781, Statement in Support of Claimed Mental Health Disorder(s) Due to an In-Service Traumatic Event(s)” 
- After a successful upload display an option to delete the file
- After a successful upload display an option to change the file
- After a successful upload display an option to add another file 

**Change file**
- If the user selects to change the file, allow them to upload a new file to replace the existing (follow steps for upload)
  
~**Delete Modal**   DONT DO THIS, this is in [another ticket ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/102579)
  - If the user selects the delete option, display an are you sure modal 
    - A11y
      - The alert is only available to users when the modal is active. 
      - When the modal is active, the rest of the page is unavailable by mouse, keyboard, touch, or screen reader.
      - When a dialog opens, focus moves to the header
 - Close action closes the modal and focus is returned to the "delete file" button
 - No action closes the modal and focus is returned to the "delete file" button
 - Yes action closes the modal and deletes the file~

**Additional Actions** 
- Return to additional action will take user back to [3.1c] Start page - in progress status
- Back action returns the user to the [3.2] Choice page
- Continue action directs the user to the [3.1c] Start page - in completed status 

### Notes
-  There is a new name for the file: “VA Form 21-0781, Statement in Support of Claimed Mental Health Disorder(s) Due to an In-Service Traumatic Event(s)” (Previous name: “VA Form 21-0781 - Statement in Support of Claim for PTSD”)
~- _**Waiting on OCTO to confirm~ - see comments below**_
     Backend type key code used to track the documents: ‘L228’ => ‘VA Form 21-0781 - Statement in Support of Claim for PTSD’
- The current feature for upload documents that is in production has a different design than what is in the design specs. Need to confirm with Design and OCTO that we are required to use the new design vs. the old and whether if we do change it, do we need to change it on the existing flow or only in our new 0781 flow. 
- The expected functionality of the upload document process should remain the same as what is in production today, unless errors are found. 
- When testing uploads founds some "quirks", one around when attempting to upload another (2nd file) where two cancel options were displayed in the same box (see comments below with screen shot)

See [[0781 PS] (4) PDF Upload Page Content](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97075)	https://github.com/department-of-veterans-affairs/va.gov-team/issues/97075 for content and style. 

[See the Planning Doc for more info](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/0781_paper_sync_plannig.md)

**Comment by lisacapaccioli** [comment]
Question to OCTO: [RE: 0781 Paper Sync - user choses to upload PDF vs. online](https://dsva.slack.com/archives/C04KW0B46N5/p1733241159031509)

**Comment by lisacapaccioli** [comment]
Found this issue of displaying two cancel options when attempting to upload a second file.
 ![Image](https://github.com/user-attachments/assets/ef98feae-1932-4386-84dd-31f463580005)

**Comment by lisacapaccioli** [comment]
[Checking](https://dsva.slack.com/archives/C0NGDDXME/p1733417345012539) with design team if the new design pattern has been implemented before. Not sure how the delete function should work. My assumption is that it's just be a remove file and show default state and a modal is no longer used.

**Comment by lisacapaccioli** [comment]
@rubensun - Can we please confirm if we use this new design for the new flow for 0781 file upload that the old design remains in the old flow? I don't want to make the assumption that it should. If it needs to change for both, we'll have to account for that work as well since it will be additional scope to the project. (cc @kylesoskin @SamStuckey)

**Comment by rubensun** [comment]
As discussed in the refinement on 12/10, to de-risk implementation, the team prefers to retain existing file upload components and use as much existing code as possible. This is because the new file upload component has not been implemented by other teams and the scope to implement the 0781 is considerable to begin with.

**Comment by lisacapaccioli** [comment]
We are going to move forward with the current designs using the latest component for file upload. If engineering discovers a risk in the solution that would impact the delivery of this feature, we'll write a risk and decision document and present our recommendations to OCTO.

**Comment by kylesoskin** [comment]
Its its own page <per research/ux/design>

It will have new v3 web component <requirement to pass staging review>

This introduces inconsistent upload experience (upload for 0781 looks one way (v3 component)m, upload for general 526 docs looks another way (old component), unless we: 

1) expand scope to upgrade current general 526 upload component to v3
2) only have uploads in 1 place <Ruben says no to this>
3) just use same old component for 0781 upload as is existing in the existing 526 general upload flow <We think VA staging review people will not let us introduce a non-v3 web component into prod)

**Comment by NB28VT** [comment]
According to @kylesoskin, David Reis has indicated the VA internal document code for this will remain 228 as with the existing Form 0781

**Comment by NB28VT** [comment]
@rubensun  @lisacapaccioli  another foil here:

Currently, the upload pages in Form 526 send a document to an endpoint that creates a "Supporting Evidence Attachment" record in the backend. This is where we store any user-uploaded evidence, which we later submit to Lighthouse when Form 526 is submitted.

It will make the most sense to do the same for an uploaded Form 0781, however, as with the other pieces of evidence the user will have the ability to remove the document when they review their form and evidence submission at the end.

Do we want to allow this? Are there any implications of that?

**Comment by lisacapaccioli** [comment]
[Here is Nathan and I's analysis regarding:](https://dsva.slack.com/archives/C04KW0B46N5/p1737734016926309?thread_ts=1737665789.330649&cid=C04KW0B46N5)

-0781 and evidence upload V3 component
OCTO feels there is value to continue to deliver on V3 component. Our discovery and custom code, wether we are first or not, will add to the overall benefit of B&C products that use the upload feature.
More information about the endpoint - payload is different, but nothing about why it’s different. A little detail will help.
   - Are there any potential risks to consider
   - What is the expected effort

Anticipated Current Deficiencies in V3 component (vs existing V2 component)
- No locked PDF support 
   - large change, additional inputs/states of component to detect and prompt for/accept a password 
   - Ticket, XL
- No flexible max_file_size support
   - trivial change
   - Ticket, S
- Flexible/Dynamic Payload
   - Need to be able to call various endpoints, with varying structures of the data
   - Ticket, M/L

Other changes required to support
- Any existing bugs/issues
   - BMT team has backlog of issues they have identified with the V3 component
   - Unknown level of effort???
   - Unknown amount of bugs/issues???
- Consult/work with other teams also trying to currently make changes to this
   - Simple forms team
    - BMT (CST) team
    - Forms Library team

Direct answers to asked questions
- More information about the endpoint - payload is different, but nothing about why it’s different. A little detail will help.
    - The current component sends data to the backend in one way, our  (and other teams/forms) existing endpoints expect data in other ways. The component needs to be refactored to be more flexible, to support sending data to arbitrary endpoints since it is a general component.
- Are there any potential risks to consider
   - Yes, we’ve outlined many of these risks
   - Adds dependency on other teams
   - Overhead in making generalized component vs component only we are using
   - Delays to overall delivery of 0781 in favor of making a generalized component for anyone to use
   - Unknown/known bugs with V3 component need addressed before release
- What is the expected effort
    - This is outlined above
    - At least 3 tickets, representing about 3 sprints worth of work (may be more or less depending on effort/priority from other teams involved in this component, and any platform requirements for forms library components such as design doc updating, Storybook updating, etc)

**Comment by lisacapaccioli** [comment]
[Lisa Capaccioli](https://dsva.slack.com/archives/C04KW0B46N5/p1738078571142969?thread_ts=1737665789.330649&cid=C04KW0B46N5)
The engineering team feels that until they know how they should proceed forward that they can’t provide an estimate at this time.
There are 4 teams all trying to do the same thing, and all teams need it ASAP. The team feels that the decision to move forward is in OCTO’s hands. Do we proceed to form a team of teams and coordinate the work together, or do we plan to do it ourselves (i.e., just DBEX-2)
Working with the others means we have dependencies on each other, but less risk to the feature. Working alone means we can forge ahead with our own requirements, but risk impacting each other’s work/code.

[Emily Theis ]((https://dsva.slack.com/archives/C04KW0B46N5/p1738081758885189?thread_ts=1737665789.330649&cid=C04KW0B46N5))(OOO 1/30-1/31)
Assuming the team's recommendation is still to go with a team of teams collaborative approach, I'm good with that.

Providing your evidence to the VA is an important task across the platform, and working with others helps us make sure that feature gets done right, even if it increases the dependencies and timeline as a tradeoff.

**Comment by kylesoskin** [comment]
Closing as old/duplicate in favor of https://github.com/department-of-veterans-affairs/va.gov-team/issues/102173

**Comment by lisacapaccioli** [comment]
Reopened this so that we could track the V3 Component that was originally planned, but descoped.
